extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -27,11 +76,69 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
    * @return {!Promise}
    */
   open(shouldAnimate = true) {
+    this.populatePlp_();
+
     return this.attachmentEl_
       .getImpl()
       .then((impl) => impl.open(shouldAnimate));
   }
 
+  /** @private */
+  populatePlp_() {
+    if (this.plpContainer_.querySelector('.amp-story-shopping-plp')) {
+      return;
+    }
+    const shoppingData = this.storeService_.get(StateProperty.SHOPPING_DATA);
+    const shoppingDataForPage = Array.from(this.shoppingTags_).map(
+      (shoppingTag) => shoppingData[shoppingTag.getAttribute('data-tag-id')]
+    );
+    this.plpContainer_.appendChild(
+      this.renderPlpTemplate_(shoppingDataForPage)
+    );
```

Comment:
Please render the template into a variable, and then append within a `mutateElement`